src/biguint.rs

```diff
@@ -520,6 +520,7 @@ impl Zero for BigUint {
     #[inline]
     fn set_zero(&mut self) {
         self.data.clear();
+        self.normalize();
```

Comment:
Hmm, the shrink condition is `len < capacity / 4`, which means `set_zero` would only keep its allocation if it started with 3 or less. That's not much if someone is trying to avoid reallocation, but I'm also not sure anyone really cares...